tables/waf_traffic_log/waf_traffic_log_mapper.go

```diff
@@ -0,0 +1,96 @@
+	var jsonBytes []byte
+
+	switch v := a.(type) {
+	case []byte:
+		jsonBytes = v
+	case string:
+		jsonBytes = []byte(v)
+	default:
+		return nil, fmt.Errorf("expected byte[] or string, got %T", a)
+	}
+
+	// decode JSON into WafTrafficLog
+	var log WafTrafficLog
+	err := unmarshalWafTrafficLog(jsonBytes, &log)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding JSON: %w; partial log: %+v", err, log)
+	}
+
+	return &log, nil
+}
+
+func unmarshalWafTrafficLog(data []byte, log *WafTrafficLog) error {
+	var temp struct {
+		Timestamp                   *int64                 `json:"timestamp"`
+		FormatVersion               *int32                 `json:"formatVersion"`
+		CaptchaResponse             *CaptchaResponse       `json:"captchaResponse"`
+		WebAclId                    *string                `json:"webAclId"`
+		TerminatingRuleMatchDetails []TerminatingRuleMatch `json:"terminatingRuleMatchDetails,omitempty"`
+		TerminatingRuleId           *string                `json:"terminatingRuleId,omitempty"`
+		TerminatingRuleType         *string                `json:"terminatingRuleType,omitempty"`
+		Action                      *string                `json:"action"`
+		HttpSourceName              *string                `json:"httpSourceName,omitempty"`
+		HttpSourceId                *string                `json:"httpSourceId,omitempty"`
+		RuleGroupList               []RuleGroup            `json:"ruleGroupList,omitempty"`
+		RateBasedRuleList           []RateBasedRule        `json:"rateBasedRuleList,omitempty"`
+		NonTerminatingMatchingRules []Rule                 `json:"nonTerminatingMatchingRules,omitempty"`
+		HttpRequest                 *HttpRequest           `json:"httpRequest,omitempty"`
+		RequestHeadersInserted      []Header               `json:"requestHeadersInserted,omitempty"`
+		Labels                      []Labels               `json:"labels,omitempty"`
+	}
+
+	// Unmarshal JSON into temporary struct
+	if err := json.Unmarshal(data, &temp); err != nil {
+		return err
+	}
+
+	// Assign values from temp struct
+	log.FormatVersion = temp.FormatVersion
+	log.WebAclId = temp.WebAclId
+	log.TerminatingRuleMatchDetails = temp.TerminatingRuleMatchDetails
+	log.TerminatingRuleId = temp.TerminatingRuleId
+	log.TerminatingRuleType = temp.TerminatingRuleType
+	log.Action = temp.Action
+	log.RuleGroupList = temp.RuleGroupList
+	log.RateBasedRuleList = temp.RateBasedRuleList
+	log.NonTerminatingMatchingRules = temp.NonTerminatingMatchingRules
+	log.HttpRequest = temp.HttpRequest
+	log.RequestHeadersInserted = temp.RequestHeadersInserted
+	log.Labels = temp.Labels
+	log.CaptchaResponse = temp.CaptchaResponse
+
+	// For a rule that triggered on SQLi detection(terminating/non-terminating) will not have HttpSourceName and HttpSourceId.
+	if *temp.HttpSourceName == "-" {
+		log.HttpSourceName = nil
+	}
+	if *temp.HttpSourceId == "-" {
+		log.HttpSourceId = nil
+	}
```

Comment:
These will by default be `nil` on `log`, however you're never actually going to set them using this approach.

Maybe move these checks above the `log` assignments and set the temp value to `nil` if it's currently `-` 
```go
if temp.HttpSourceName != nil && *temp.HttpSourceName == "-" {
  temp.HttpSourceName = nil
}
```
and then in `log` assignments you can `log.HttpSourceName = temp.HttpSourceName` etc.

Reply:
Done!